chore: Refactors data-source tests together with resource tests and updates import docs (#3292)
mongodb/terraform-provider-mongodbatlas: docs/resources/database_user.md

Database users can be imported using project ID, username, and auth database name in the format:

> 1. `project_id`-`username`-`auth_database_name` Might not work if `-` are used in both `username` and `auth_database_name`. For example `my-username` and `my-db` should use (2).

Reviewer: `might not` is a bit vague IMO.

Author: Changed in  c6de2c1a3af2689bcb49fe3f9b17aed07fe8086d

> 1. `project_id`-`username`-`auth_database_name` Might not work if `-` are used in both `username` and `auth_database_name`. For example `my-username` and `my-db` should use (2).

Reviewer 2: ```suggestion
1. `project_id`-`username`-`auth_database_name` Doesn't  work if `-` is used in both the `username` and the `auth_database_name`. For example `my-username` and `my-db` should use (2).
```

Author: Thank you changed in c6de2c1a3af2689bcb49fe3f9b17aed07fe8086d

> 2. `project_id`/`username`/`auth_database_name` Works when neither `username` or `auth_database_name` use `/`.

Reviewer 2: ```suggestion
2. `project_id`/`username`/`auth_database_name` Works when neither `username` nor `auth_database_name` use `/`.
```

Author: Thank you changed in c6de2c1a3af2689bcb49fe3f9b17aed07fe8086d
